lmtp_server_test.go

```diff
@@ -195,3 +200,103 @@ func TestServer_LMTP_DuplicatedRcpt(t *testing.T) {
 		t.Fatal("Invalid number of sent messages:", be.messages, be.anonmsgs)
 	}
 }
+
+func TestServer_LMTP_NetworkDefaultBehaviorUnix(t *testing.T) {
+	be := new(backend)
+	s := smtp.NewServer(be)
+	s.Domain = "localhost"
+	s.AllowInsecureAuth = true
+
+	s.Network = ""
+	s.LMTP = true
+	s.Addr = "./testsocket"
+
+	defer os.Remove("./testsocket")
+	go s.ListenAndServe()
+
+	var err error
+	var c net.Conn
+	for i := 0; i < 5; i++ {
+		c, err = net.Dial("unix", s.Addr)
+		if err == nil {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	scanner := bufio.NewScanner(c)
+	scanner.Scan()
+	if scanner.Text() != "220 localhost ESMTP Service Ready" {
+		t.Fatal("Invalid greeting:", scanner.Text())
+	}
+}
+
+func TestServer_LMTP_TCP(t *testing.T) {
+	be := new(backend)
+	s := smtp.NewServer(be)
+	s.Domain = "localhost"
+	s.AllowInsecureAuth = true
+
+	testPort := rand.Int31n(65535-1024) + 1024
```

Comment:
This is fragile because the port could be in use. Use port zero instead.

Reply:
Totally agree, do you know a way to get the actually used port then? The other test code uses Serve with a custom listener, this here tests logic in ListenAndServe and can't use that

Reply:
Actual listening port should be available via net.Listener.Addr().